src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTableUI.java

```diff
@@ -178,7 +181,9 @@ private void updateStyle(JTable c) {
             if (rowHeight != null) {
                 LookAndFeel.installProperty(table, "rowHeight", rowHeight);
             }
-            boolean showGrid = style.getBoolean(context, "Table.showGrid", true);
+            showHorizLines = table.getShowHorizontalLines();
+            showVertLines = table.getShowVerticalLines();
+            showGrid = style.getBoolean(context, "Table.showGrid", true);
             if (!showGrid) {
                 table.setShowGrid(false);
             }
```

Comment:
What if the grid isn't displayed and `showGrid` is set to `true`? Then the value of `showGrid` isn't respected again.